website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,187 @@
+```
+
+Run the app
+
+```sh
+npm run dev
+```
+
+And you should see something like this:
+
+![chatbot UI](./sveltekit-chatbot.png)
+
+This is a simple Sveltekit app with a chatbot UI. It doesn't really matter if you've never used Sveltekit before. It's similar to Next.js or Nuxt or SolidStart, where your file structure defines your routes.
+
+The most important file for this tutorial is the `src/routes/api/chat/+server.js` file.
+
+It creates an API endpoint at `/api/chat`. When your frontend sends a POST request to `/api/chat`, this is the code that handles it.
+
+```javascript
+import { createOpenAI } from "@ai-sdk/openai";
+import { streamText } from "ai";
+import { env } from "$env/dynamic/private";
+
+const openai = createOpenAI({
+    apiKey: env.OPENAI_API_KEY ?? "",
+});
+
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+
+export const POST = async ({ request }) => {
+    const { messages } = await request.json();
+    const variant = variants["variant1"];
+    const result = await streamText({
+        model: openai(variant.model),
+        messages,
+        max_tokens: variant.max_tokens,
+        temperature: variant.temperature,
+    });
+    return result.toDataStreamResponse();
+};
+```
+
+This file is doing a few key things:
+
+1. It sets up our OpenAI client with an API key.
+2. It defines different AI model configurations in the `variants` object.
+3. The `POST` function handles incoming chat requests.
+
+When a request comes in, it:
+
+-   Extracts the messages from the request body
+-   Selects a variant (currently hardcoded to 'variant1')
+-   Uses the OpenAI API to generate a response
+-   Streams the response back to the client
+
+The `streamText` function part of some utilities provided by Vercel's AI SDK, which helps deal with real-time streaming of the AI's responses and other AI-related tasks.
+
+## Creating a feature flag with AI variants
+
+Instead of hardcoding `variant1`, we want to use feature flags to dynamically choose which AI model to use. This will let us easily switch between models, test different configurations, or even do some A/B testing.
+
+To implement this, we'll need to:
+
+1. Set up a feature flag provider (we'll use Unleash).
+2. Replace our static `variants` selection with feature flag calls.
+3. Use the feature flag in our code to determine which AI model and settings to use for each request.
+
+### Install a local feature flag provider
+
+In this section, we’ll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you’ll need to update the code accordingly. The basic steps will probably be the same.
```

Comment:
```suggestion
In this section, we’ll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you’ll need to update the code accordingly.
```